### PR TITLE
Add survey list sorting

### PR DESCRIPTION
**Pull request by keithjgrant**
##### SUMMARY
* Adds up/down buttons to re-order survey questions.
* Adds TemplateSurvey component to aid with routing & fetching of survey data

Addresses #6182 

Remaining TODO (for separate PR?): show all loading states in a meaningful way to the user after toggling, re-ordering, or deleting

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui_next/src/screens/Template/shared/SurveyList.jsx [hunk 1]
```diff
@@ -1,129 +1,75 @@
...
-  }, [toggleError]);
-
   const handleSelectAll = isSelected => {
     setSelected(isSelected ? [...questions] : []);
   };
 
   const handleSelect = item => {
-    if (selected.some(s => s.id === item.id)) {
-      setSelected(selected.filter(s => s.id !== item.id));
+    if (selected.some(q => q.variable === item.variable)) {
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Template/shared/SurveyList.jsx:32]: We feel pretty confident that variable is unique?

File: awx/ui_next/src/screens/Template/shared/SurveyList.orig.jsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
+import React, { useEffect, useCallback, useState } from 'react';
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Template/shared/SurveyList.orig.jsx]: Looks like this file needs to be removed?

File: awx/ui_next/src/screens/Template/shared/SurveyList.jsx (hunk 1 above)
Comment (mabashian) [reply awx/ui_next/src/screens/Template/shared/SurveyList.jsx:32]: The `variable` field should be unique - you can't have two questions with the same `variable` value in a survey

**Comment by keithjgrant** [comment]
Note: this PR has some Survey files living in `Template/shared` that need to be moved into something like `Template/TemplateSurvey` but I'm holding off on that while other survey changes are in flight

File: awx/ui_next/src/util/useRequest.js [hunk 3]
```diff
@@ -8,11 +8,12 @@ import {
 
 /*
  * The useRequest hook accepts a request function and returns an object with
- * four values:
+ * six values:
```
Comment (marshmalien) [inline awx/ui_next/src/util/useRequest.js]: five?

File: awx/ui_next/src/util/useRequest.js [hunk 4]
```diff
@@ -34,48 +35,71 @@ export default function useRequest(makeRequest, initialValue) {
...
     error,
     isLoading,
-    request: useCallback(async () => {
-      setIsLoading(true);
-      try {
-        const response = await makeRequest();
-        if (isMounted.current) {
-          setResult(response);
+    request: useCallback(
+      async (...args) => {
```
Comment (marshmalien) [inline awx/ui_next/src/util/useRequest.js:39]: Yes, I'm glad you extended this hook to accept arguments.

File: awx/ui_next/src/screens/Template/shared/SurveyList.test.jsx [hunk 5]
```diff
@@ -1,91 +1,69 @@
...
-  beforeEach(() => {
-    JobTemplatesAPI.readSurvey.mockResolvedValue({
-      data: {
-        name: 'Survey',
-        description: 'description for survey',
-        spec: [{ question_name: 'Foo', type: 'text', default: 'Bar' }],
-      },
-    });
-  });
+  // beforeEach(() => {
```
Comment (marshmalien) [inline awx/ui_next/src/screens/Template/shared/SurveyList.test.jsx]: Did we mean to remove this beforeEach()?

**Review by marshmalien — APPROVED** [review]
Great work @keithjgrant and @AlexSCorey . The survey list reordering functionality is fun to play with, and I’m looking forward to using the latest additions to useRequest!

**Pull request merged**